code/modules/clothing/shoes/magboots.dm

```diff
@@ -18,14 +18,19 @@
 	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/item/clothing/shoes/magboots/water_act(volume, temperature, source, method)
+	. = ..()
+	if(magpulse && slowdown_active > SHOES_SLOWDOWN)
+		slowdown = slowdown_active
+
 /obj/item/clothing/shoes/magboots/atmos
 	desc = "Magnetic boots, made to withstand gusts of space wind over 500kmph."
 	name = "atmospheric magboots"
 	icon_state = "atmosmagboots0"
 	magboot_state = "atmosmagboots"
 	gustprotection = TRUE
 
-/obj/item/clothing/shoes/magboots/attack_self(mob/user)
+/obj/item/clothing/shoes/magboots/attack_self(mob/user, forced = FALSE)
```

Comment:
This should probably be encapsulated into its own proc, instead of adding `forced` to attack_self. Something like `toggle_magpulse(mob/user, forced = FALSE)`. Then this can be called from both attack_self and anywhere else toggling of the magboots is needed

Reply:
done